.github/workflows/clang-tidy.yml

```diff
@@ -0,0 +1,49 @@
+name: Clang Tidy
+
+on:
+  schedule:
+    - cron: "0 1 * * *"
+  push:
```

Comment:
Do we need to worry about the push section, I assumed we had pushing to the main branch disabled?

Reply:
Thanks. I’ll review the other actions too. They should be consistent.

Reply:
This is consistent with other CI checks. We should clean up (or not) in a follow-up PR.